gamemodes/terrortown/gamemode/client/cl_eventpopup.lua

```diff
@@ -1,53 +1,150 @@
+		text = "Well, hello there! This is a fancy popup with some special information. The text can be also multiline, how fancy! Ugh, I could add so much more text if I'd had any ideas..."
+	},
+	iconTable = {},
+	time = CurTime() + 5
+}
+
+EPOP = EPOP or {}
+
+EPOP.messageQueue = EPOP.messageQueue or {}
 
 ---
 -- Adds a popup message to the @{EPOP}
 -- @param string|table title The title of the popup that will be displayed in large letters (can be a table with `text` and `color` attribute)
--- @param string|table text An optional description that will be displayed below the title (can be a table with `text` and `color` attribute)
--- @param number disp_time The render duration of the popup
--- @param table icon_tbl An optional set of icon materials that will be rendered below the popup
+-- @param [opt]string|table subtitle An optional description that will be displayed below the title (can be a table with `text` and `color` attribute)
+-- @param [default=4]number displayTime The render duration of the popup
+-- @param [opt]table iconTable An optional set of icon materials that will be rendered below the popup
+-- @param [default=true]boolean blocking If this is false, this message gets instantly replaced if a new message is added
+-- @return string Returns a unique id generated for this message
 -- @realm client
-function EPOP:AddMessage(title, text, disp_time, icon_tbl)
-	self.msg = {
-		title = {},
-		text = {}
-	}
+function EPOP:AddMessage(title, subtitle, displayTime, iconTable, blocking)
+	local id = "epop_unique_id_" .. tostring(CurTime() * 10000)
 
-	if isstring(title) then
-		self.msg.title.text = title
-	else
-		self.msg.title = title
+	local queueSize = #self.messageQueue
+
+	-- delete previous message if it is nonblocking
+	if queueSize > 0 and not self.messageQueue[queueSize].blocking then
+		table.remove(self.messageQueue, queueSize)
+
+		queueSize = queueSize - 1
 	end
 
-	if isstring(text) then
-		self.msg.text.text = text
-	else
-		self.msg.text = text
+	-- add the new message to the queue
+	self.messageQueue[queueSize + 1] = {
+		id = id,
+		title = isstring(title) and {text = title} or title,
+		subtitle = isstring(subtitle) and {text = subtitle} or subtitle,
+		displayTime = displayTime or 4,
+		iconTable = iconTable or {},
+		blocking = blocking == nil and false or blocking
+	}
+
+	-- the new element is the first one in the list and should be therefore activated
+	if queueSize == 0 then
+		self:ActivateMessage()
 	end
 
-	self.msg.icon_tbl = icon_tbl or {}
-	self.msg.time = CurTime() + (disp_time or 4)
+	return id
+end
+
+---
+-- Activates a new message by setting up all internal values
+-- @internal
+-- @realm client
+function EPOP:ActivateMessage()
+	if #self.messageQueue == 0 then return end
+
+	local elem = self.messageQueue[1]
+
+	elem.time = CurTime() + elem.displayTime
+
+	-- register a timer to remove the message
+	timer.Create(TIMER_IDENTIFIER, elem.displayTime, 1, function()
+		EPOP:RemoveMessageByIndex(1)
```

Comment:
This should rather be RemoveMessage instead, as this timer should only be allowed to delete this exact element, to reduce sideeffects